<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8">
</head>
<body>
<aside style="display:none;">connaissances nécessaire: objet, constructeur</aside>
thème: objet<br>
niveau: 3<br><br>
<b>code:</b><br>
<textarea rows="20" cols="100" id="question">
function Constructeur(valeur){
	this.f = function(valeur){
		this.get = function(valeur){
			return this.calcul();
		};
	};
	this.calcul = function(){
		return valeur;
	};
}

var obj1 = new Constructeur(1),
	obj2 = new obj1.f(2);
alert(obj2.get(3));

</textarea>

<hr>
<b>Question:</b> Qu'est-ce qui s'affiche si on exécute ce code?
<ul>
	<li>1</li>
	<li>2</li>
	<li>3</li>
	<li>undefined</li>
	<li class="bonneReponse">rien, une erreur est survenue</li>
</ul>
<br>
<hr>
<br>
<b>Résultat</b> <i>essayez vous-même</i>:
<br>
<input type="button" id="input" value="tester" onclick="test()">


<details style="margin-top:3em;border:1px solid black;display:block;">
	<summary>Explication</summary>
	
</details>

<script>
function test(){
	var code = document.getElementById("question").value;
	if(!/alert\(/.test(code)){
		code = code.replace(/(?:^|[\r\n])(?:[\r\n]|$)/g,"").replace(/(^|[\n\r;])([^;\r\n]+);?$/,"$1alert($2);");
	}
	try{
		var f=new Function(code);
	}catch(e){
		alert("ERROR while creating test\n"+e.message);
	}
	if(f){
		try{
			f();
		}catch(e){
			alert("ERROR\n"+e.message);
		}
	}
}
</script>

</body>
</html>
